src/main/java/net/mcreator/ui/modgui/AttributeGUI.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * MCreator (https://mcreator.net/)
+ * Copyright (C) 2012-2020, Pylo
+ * Copyright (C) 2020-2024, Pylo, opensource contributors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package net.mcreator.ui.modgui;
+
+import net.mcreator.element.types.Attribute;
+import net.mcreator.ui.MCreator;
+import net.mcreator.ui.component.JMinMaxSpinner;
+import net.mcreator.ui.component.util.PanelUtils;
+import net.mcreator.ui.help.HelpUtils;
+import net.mcreator.ui.init.L10N;
+import net.mcreator.ui.minecraft.EntityListField;
+import net.mcreator.ui.validation.AggregatedValidationResult;
+import net.mcreator.ui.validation.ValidationGroup;
+import net.mcreator.ui.validation.Validator;
+import net.mcreator.ui.validation.component.VTextField;
+import net.mcreator.ui.validation.validators.TextFieldValidator;
+import net.mcreator.util.StringUtils;
+import net.mcreator.workspace.elements.ModElement;
+
+import javax.swing.*;
+import java.awt.*;
+
+public class AttributeGUI extends ModElementGUI<Attribute> {
+	
+	private final VTextField name = new VTextField(20);
+	private final JSpinner defaultValue = new JSpinner(new SpinnerNumberModel(0.0, -Double.MAX_VALUE, Double.MAX_VALUE, 1.0));

```

Comment:
While this may work with double, it wouldn't with ints, so for sake of code-correctnes, use MIN_VALUE here instead of flipping sign bit on max value

Reply:
doing this
![image](https://github.com/user-attachments/assets/e073dac2-76d7-463c-9efa-81ecb74828ee)
results in this
![image](https://github.com/user-attachments/assets/0dddc0ae-0214-418c-83a6-d7d2153ed5b1)

Reply:
this is because MIN_VALUE is the smallest possible non-zero value for some reason

Reply:
Interesting, didn't know floating point numbers behave like that. All good in this case :)